internal/syncer/syft_repo_scan.go

```diff
@@ -0,0 +1,84 @@
+package syncer
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os/exec"
+
+	"github.com/jackc/pgx/v4"
+	libgit2 "github.com/libgit2/git2go/v33"
+	"github.com/mergestat/fuse/internal/db"
+)
+
+// handleSyftRepoScan executes `syft {git-repo} -f json` for a repo
+// and inserts the output JSON into the DB
+func (w *worker) handleSyftRepoScan(ctx context.Context, j *db.DequeueSyncJobRow) error {
+	l := w.loggerForJob(j)
+
+	// indicate that we're starting query execution
+	if err := w.formatBatchLogMessages(ctx, SyncLogTypeInfo, j, jobStatusTypeInit); err != nil {
+		return fmt.Errorf("log messages: %w", err)
+	}
+
+	tmpPath, cleanup, err := w.createTempDirForGitClone(j)
+	if err != nil {
+		return fmt.Errorf("temp dir: %w", err)
+	}
+	defer cleanup()
+
+	var ghToken string
+	if ghToken, err = w.fetchGitHubTokenFromDB(ctx); err != nil {
+		return err
+	}
+
+	var repo *libgit2.Repository
+	if repo, err = w.cloneRepo(ctx, ghToken, j.Repo, tmpPath, false, j); err != nil {
+		return fmt.Errorf("git clone: %w", err)
+	}
+	defer repo.Free()
+
+	cmd := exec.CommandContext(ctx, "syft", ".", "-o", "json")
+	cmd.Dir = tmpPath
+
+	var output []byte
+	if output, err = cmd.Output(); err != nil {
+		if exitErr, ok := err.(*exec.ExitError); ok {
+			w.logger.Err(exitErr).Str("stderr", string(exitErr.Stderr)).Msgf("error running syft repo scan")
+		}
+		return fmt.Errorf("running syft scan: %w", err)
+	}
+
+	var tx pgx.Tx
+	if tx, err = w.pool.BeginTx(ctx, pgx.TxOptions{}); err != nil {
+		return fmt.Errorf("begin tx: %w", err)
+	}
+	defer func() {
+		if err := tx.Rollback(ctx); err != nil {
+			if !errors.Is(err, pgx.ErrTxClosed) {
+				w.logger.Err(err).Msgf("could not rollback transaction")
+			}
+		}
+	}()
+
+	if _, err := tx.Exec(ctx, "DELETE FROM syft_repo_scans WHERE repo_id = $1;", j.RepoID.String()); err != nil {
+		return fmt.Errorf("exec delete: %w", err)
+	}
+
```

Comment:
We should send a batch  to get the log  of the delete op :) 
`if err := w.sendBatchLogMessages(ctx, []*syncLog{{
		Type:            SyncLogTypeInfo,
		RepoSyncQueueID: j.ID,
		Message:         fmt.Sprintf("removed %d row(s) from syft_repo_scans", r.RowsAffected()),
	}}); err != nil {
		return err
	}`